File: Source/Aladdin.Java/src/aladdin/remoting/IBackgroundTask.java
package aladdin.remoting;
import aladdin.*; 

///////////////////////////////////////////////////////////////////////////
// Задание, выполняемое удаленно
///////////////////////////////////////////////////////////////////////////
public interface IBackgroundTask extends IRefObject
{
    // признак выполнения и идентификатор потока
    boolean isBusy(); long threadID();  
    // признак отмены задания 
    boolean cancellationPending(); 

    // запустить и отменить задание
    void start(Object argument) throws Exception; void cancel();
    
    // обработчик событий
    IBackgroundHandler localHandler(); 
    
    // указать прогресс операции
    void reportProgress(int percentProgress, Object userState);
}

File: Source/Aladdin/Remoting/IBackgroundTask.cs
namespace Aladdin.Remoting
{
    ///////////////////////////////////////////////////////////////////////////
    // Задание, выполняемое удаленно
    ///////////////////////////////////////////////////////////////////////////
    public interface IBackgroundTask : IRefObject
    {
        // признак выполнения и идентификатор потока
        bool IsBusy { get; } int ThreadID { get; } 
        // признак отмены задания 
        bool CancellationPending { get; }

        // обработчик событий
        IBackgroundHandler LocalHandler { get; }

        // запустить и отменить задание
        void Start(object argument); void Cancel();

        // указать прогресс операции (для удаленного потока)
        void ReportProgress(int percentProgress, object userState);
    }
}
